doc/en/deprecations.rst

```diff
@@ -252,6 +252,47 @@ or ``pytest.warns(Warning)``.
 
 See :ref:`warns use cases` for examples.
 
+
+Returning non-None value in test functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. deprecated:: 7.0
```

Comment:
Like you say, 7.0 has already been released, so this should be

```suggestion
.. deprecated:: 7.2
```

I believe? (not 8.0 I think - from what I can see, we deprecated things in new feature releases before)